### ISSUE TITLE
Bug: sort-imports generates wrong results

### ISSUE THREAD

**Issue by ModyQyW** [azat-io/eslint-plugin-perfectionist#80]
### Describe the bug

ESLint config:

```js
// .eslintrc.cjs
module.exports = {
  plugins: ["perfectionist"],
  rules: {
    "perfectionist/sort-imports": [
      "error",
      {
        groups: [
          "builtin",
          "external",
          "internal",
          "parent",
          "siblings",
          "side-effect",
          "index",
          "object",
          "style",
          "builtin-type",
          "external-type",
          "internal-type",
          "parent-type",
          "sibling-type",
          "index-type",
          "unknown",
        ],
        "internal-pattern": ["~/**", "~~/**", "@/**", "@@/**"],
        "newlines-between": "ignore",
      },
    ],
  },
  env: {
    es2024: true,
  },
  parserOptions: {
    sourceType: "module",
  },
};

```

JavaScript file: 

```js
// test.js
import debug from 'debug';
import { test } from './helpers';
```

ESLint reports an error.

<img width="805" alt="image" src="https://github.com/azat-io/eslint-plugin-perfectionist/assets/31850793/a9d0eeae-fa55-496a-88a7-c940940b7987">

However I think `debug` should come before `./helpers` according to [the docs](https://eslint-plugin-perfectionist.azat.io/rules/sort-imports#groups) with these ESLint config.

<img width="588" alt="image" src="https://github.com/azat-io/eslint-plugin-perfectionist/assets/31850793/6e41675f-6801-4b25-95ad-5f5fd0bcc643">

### Code example

test.js: 

```js
// test.js
import debug from 'debug';
import { test } from './helpers';

```

helpers.js: 

```js
// helpers.js
export function test() {}
```

.eslintrc.cjs

```js
// .eslintrc.cjs
module.exports = {
  plugins: ["perfectionist"],
  rules: {
    "perfectionist/sort-imports": [
      "error",
      {
        groups: [
          "builtin",
          "external",
          "internal",
          "parent",
          "siblings",
          "side-effect",
          "index",
          "object",
          "style",
          "builtin-type",
          "external-type",
          "internal-type",
          "parent-type",
          "sibling-type",
          "index-type",
          "unknown",
        ],
        "internal-pattern": ["~/**", "~~/**", "@/**", "@@/**"],
        "newlines-between": "ignore",
      },
    ],
  },
  env: {
    es2024: true,
  },
  parserOptions: {
    sourceType: "module",
  },
};

```

package.json:

```json
{
  "name": "myapp",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "eslint": "^8.51.0",
    "eslint-plugin-perfectionist": "^2.1.0"
  },
  "dependencies": {
    "debug": "^4.3.4"
  }
}

```

### ESLint version

8.51.0

### ESLint Plugin Perfectionist version

2.1.0

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Fixed in a22eaf6d444b299c1f7255cdba878bdffaefefe3.
I'll create a release tomorrow.

Also, there is an error in your example. We don't have a `sibling` group, only a `sibling`.

**Comment by ModyQyW** [comment]
Thanks!

**Comment by azat-io** [comment]
Released in [v2.2.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v2.2.0).
